python_r_markdown.Rmd

```diff
@@ -0,0 +1,186 @@
+
+```{r}
+library(reticulate)
+use_python("/home/karan/anaconda3/bin/python3.5")
+```
+
+Objects can be passed easily between R and Python code chunks. To access a Python object in R, prefix it with `py$`. To access an R object in Python, prefix it with `r.` 
+
+```{r}
+#R
+string_list = c("R", "and", "Python")
+```
+
+When an R vector is accessed from a Python code chunk, it automatically becomes a list. The reverse is also true.
+
+```{python}
+#Python
+sentence = " ".join(r.string_list)
+```
+
+```{r}
+#R
+print(py$sentence)
+print(class(py$sentence))
+```
+
+Other datatype transformations that occur when objects are passed between R and Python:\
+\
+
+<table>
+<thead>
+<tr>
+<th>R</th>
+<th>Python</th>
+<th>Examples</th>
+</tr>
+</thead><tbody>
+<tr>
+<td>Single-element vector&nbsp;&nbsp;</td>
+<td>Scalar</td>
+<td><code>1</code>, <code>1L</code>, <code>TRUE</code>, <code>"foo"</code></td>
+</tr>
+<tr>
+<td>Multi-element vector</td>
+<td>List</td>
+<td><code>c(1.0, 2.0, 3.0)</code>, <code>c(1L, 2L, 3L)</code></td>
+</tr>
+<tr>
+<td>List of multiple types</td>
+<td>Tuple</td>
+<td><code>list(1L, TRUE, "foo")</code></td>
+</tr>
+<tr>
+<td>Named list</td>
+<td>Dict</td>
+<td><code>list(a = 1L, b = 2.0)</code>, <code>dict(x = x_data)</code></td>
+</tr>
+<tr>
+<td>Matrix/Array</td>
+<td>NumPy ndarray</td>
+<td><code>matrix(c(1,2,3,4), nrow = 2, ncol = 2)</code></td>
+</tr>
+<tr>
+<td>Data Frame</td>
+<td>Pandas DataFrame&nbsp;&nbsp;</td>
+<td><code>data.frame(x = c(1,2,3), y = c("a", "b", "c"))</code></td>
+</tr>
+<tr>
+<td>Function</td>
+<td>Python function</td>
+<td><code>function(x) x + 1</code></td>
+</tr>
+<tr>
+<td>Raw</td>
+<td>Python bytearray</td>
+<td><code>as.raw(c(1:10))</code></td>
+</tr>
+<tr>
+<td>NULL, TRUE, FALSE&nbsp;</td>
+<td>None, True, False</td>
+<td><code>NULL</code>, <code>TRUE</code>, <code>FALSE</code></td>
+</tr>
+</tbody>
+</table>
+\
+ <font size="1">Table source: https://cran.r-project.org/web/packages/reticulate/vignettes/calling_python.html</font>
+\
+\
+If all you need is a couple of lines of Python (or you're writing an R script instead of Markdown), an alternative is the `py_run_string()` function, or the mostly equivalent `py_eval()`. `py_eval()` returns the result of evaluating the expression as opposed to the dictionary associated with the code execution. Both belong to the `reticulate` package.
+
+```{r}
+#R
+py_run_string("x = ['short', 'order']")
+print(py$x)
+```
+You can also run a Python script with `py_run_file('script.py')`. `source_python('script.py')` is equivalent to importing a Python script. It makes all of its public objects (such as functions) available in the R environment.
+\
+\
+Here's an example that passes a dataframe from R to Python and uses `py_eval()`:
+```{r}
+#R
+data("mtcars")
+py_eval("r.mtcars.describe()")
```

Comment:
Error in py_eval_impl(code, convert) : 
  AttributeError: 'dict' object has no attribute 'describe'

Reply:
Hi, I'm not sure what to change because it works fine for me. One thing is that the Rmd needs to be knitted for any of the code segments to run, so could that be an issue? The HTML should come out fine as it does for me.

In my code 'mtcars' gets passed to Python as a dataframe, so the describe() function works. I don't know why it's a dictionary here. Any ideas?

Reply:
If it works fine locally, then I don't think you need to change anything. I will label it as a `bug` but don't worry about it. We will look into it later. Thanks for your explanation!

Reply:
I'm getting the identical bug locally -- it's not finding pandas and seaborn. These probably need to be installed like the R libraries.

Reply:
Also, this line won't work here: `use_python("/home/karan/anaconda3/bin/python3.5")`